lightning-invoice/src/utils.rs

```diff
@@ -250,12 +256,36 @@ where
 				cltv_expiry_delta: MIN_CLTV_EXPIRY_DELTA,
 				htlc_minimum_msat: None,
 				htlc_maximum_msat: None,});
+		}
+
+		phantom_hints.push(route_hints);
+	}
+
+	// We have one vector per phantom node involved in creating the invoice. To distribute the
+	// hints across our nodes we add one hint from each node in turn until no node has any hints
+	// left (if one node has more hints than any other, these will accumulate at the end of the
+	// vector).
+	let mut invoice_hints :Vec<RouteHint> = Vec::new();
```

Comment:
I suspect there is a smarter way to do this with iterators, but couldn't find an elegant one without using [itertools interleave](https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.interleave) (not currently a dependency?), and even that doesn't work well for > 2 sets. New to rust, so certainly open to better suggestions!

Reply:
Not sure if there is an easy way to do this without writing a custom `Interleave` iterator.

Regardless, we shouldn't clone anything we don't ultimately `take`. So at very least we should push references into a `Vec<&RouteHint>` and later do `invoice_hints.into_iter().take(3).cloned().collect()`.